Added documentation for staging/deployment process (#435)
OpenUpSA/static-budget-portal: README.md

git ls-files -v|grep '^h'| sed 's/^h//' | xargs git update-index --no-assume-unchanged
```

## Data generation

The static-budget-portal is the front-end (static site) of Vulekamali and does not generate or retrieve data.

> The static-budget-portal requests data from the [Data Manager](https://github.com/OpenUpSA/budget-portal) 

Reviewer: ```suggestion
The static-budget-portal build script requests data from the [Data Manager](https://github.com/OpenUpSA/budget-portal) 
```

and stores it locally in the `_data` directory, which it then uses to build the static site via Jekyll.

> That means the actual data being served in production is managed by Git.

Reviewer: could say "stored in the git repository alongside the code" instead of "managed by git"

Production data exists in the `master` branch of static-budget-portal.

> The static-budget-portal can be pointed towards any one of the following Data Manager states:

Reviewer: ```suggestion
The static-budget-portal build script can be pointed towards any one of the following Data Manager instances:
```

- Check [Travis CI](https://travis-ci.org) to see the status of the build and if there are any errors
- When Travis has completed, pull the branch (`..-data-build`) again to fetch the new data Travis built
for us
- Merge `..-data-build` into `..-ui-build`
- Serve the site locally using Jekyll to confirm that the data generated by travis during staging is correct, and that
the front-end renders correctly.
- If all is well, you should be clear to produce and deploy production data.

#### Deploy to production

Before deploying to production, ensure you've been through the Staging workflow described above.

- Checkout the existing `..-data-build` branch you've been working with
- Commit this branch to origin with `[ci]` (no staging) to build data against production
- If you've had any additional UI changes, follow step 7 from "Staging workflow"
- If all is well, open a pull request on Github to merge `..-ui-build` into `..-data-build`
    - In theory this PR should only show code changes to the UI
    - This PR should include screenshots of a local Jekyll server with the latest code/data, of the specific changes made
- Open a Github PR to merge `..-data-build` into `master` 

>     - This will kick off a Travis CI job, but it will not build the site (should last ~30 sec)

Reviewer: ```suggestion
    - This will kick off a Travis CI job, but it will not build the site data again (should last ~30 sec)
```
